### PR TITLE
Create a Jar version to use the project with args[] straight from cmd

### PR DESCRIPTION
**Pull request by shahart**
From Maven:  `java -jar target/youtubeDownload-1.0-SNAPSHOT.jar https://www.youtube.com/watch?v=3EeZVwUrT6c`
From Gradle: `java -jar build\libs\youtubeDownload.jar https://www.youtube.com/watch?v=3EeZVwUrT6c`

### CODE REVIEW

File: youtubeDownload/build.gradle [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        attributes "Main-Class": "org.example.Main"
+    }
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation('junit:junit:4.13.2')
```
Comment (shahart) [inline youtubeDownload/build.gradle:17]: Kind of "nice to have".
(can do junit Jupiter of course)

File: youtubeDownload/src/main/java/org/example/Downloader.java [hunk 2]
```diff
@@ -19,19 +23,21 @@ public static File downloadAudio(String link, File outputPath) throws IOExceptio
...
+        }
         return null;
     }
 
-
-
-
+    public static void extractYtDlp() throws IOException {
+        Files.copy(Objects.requireNonNull(Downloader.class.getClassLoader().getResourceAsStream("yt-dlp.exe")),
+                Paths.get("resources", "yt-dlp.exe"),
```
Comment (shahart) [inline youtubeDownload/src/main/java/org/example/Downloader.java:39]: Extract yt-dlp outside of the Jar, to be accessed as before

File: youtubeDownload/src/main/java/org/example/FileUtils.java [hunk 3]
```diff
@@ -21,7 +19,8 @@ public static Object runScript(String cmd, String path) throws IOException, Inte
         pb.directory(new File(path));
         pb.redirectErrorStream(true);
         Process process = pb.start();
-        flushInputStreamReader(process, false);
+        String output = flushInputStreamReader(process, false);
+        System.out.println(output);
```
Comment (shahart) [inline youtubeDownload/src/main/java/org/example/FileUtils.java:23]: log(4j) output is necessary to see if the download went well.

File: youtubeDownload/src/test/java/org/example/FileUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package org.example;
+
+import org.junit.*;
+
+public class FileUtilsTest {
+
+    @Test
+    public void testHasSolution() throws Exception {
+        Assert.assertSame("foo was found somehow", Boolean.FALSE, FileUtils.runScript("dir foo", "."));
```
Comment (shahart) [inline youtubeDownload/src/test/java/org/example/FileUtilsTest.java:9]: This is just to have some base for extra test.
`foo` denotes here an invalid file,

File: youtubeDownload/target/classes/org/example/Downloader.class
Comment (shahart) [inline youtubeDownload/target/classes/org/example/Downloader.class:1]: binary artifacts should not be part of the commits,
same for resourdes/yt-dlp.exe - we can use HTTP-client to grab the exe from the web.

**Review by OMGerEDU — APPROVED** [review]

**Comment by OMGerEDU** [comment]
Thanks alot @shahart for your contribution!

**Pull request merged**